core/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -177,6 +178,9 @@ public IndexNameExpressionResolver(Settings settings) {
         final Set<Index> concreteIndices = new HashSet<>(expressions.size());
         for (String expression : expressions) {
             AliasOrIndex aliasOrIndex = metaData.getAliasAndIndexLookup().get(expression);
+            if (context.getOptions().ignoreAliases() && aliasOrIndex.isAlias()) {
+                aliasOrIndex = null;
+            }
             if (aliasOrIndex == null) {
```

Comment:
I'd change change this to `if (aliasOrIndex == null || (aliasOrIndex.isAlias() && context.getOptions().ignoreAliases()))` and remove the new if above